c7n/resources/ami.py

```diff
@@ -609,3 +609,72 @@ def process(self, resources, event=None):
                     continue
                 results.extend(f.result())
         return results
+
+
+@AMI.filter_registry.register('image-attribute')
+class ImageAttribute(ValueFilter):
+    """AMI Image Value Filter on a given image attribute.
+
+    Filters AMI's with the given AMI attribute
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: ami-unused-recently
+                resource: ami
+                filters:
+                  - type: image-attribute
+                    attribute: lastLaunchedTime
+                    key: "Value"
+                    op: gte
+                    value_type: age
+                    value: 30
+    """
+
+    valid_attrs = (
+        'kernel',
```

Comment:
Worth noting that I intentionally left out "description" here.  I guess technically someone could filter on regex for description and we could add it for completeness, but it seems unlikely.

Reply:
I went ahead and added it in for completeness. 🤷